test/e2e/framework/globalingressips.go

```diff
@@ -0,0 +1,93 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var globalIngressIPGVR = &schema.GroupVersionResource{
+	Group:    "submariner.io",
+	Version:  "v1",
+	Resource: "globalingressips",
+}
+
+func (f *Framework) AwaitGlobalnetIP(cluster ClusterIndex, name, namespace string) string {
```

Comment:
You should have an `AwaitGlobalIngressIP` which returns the object itself. You may need it for tests that test different conditions. I used just IP coz that is more relevant for LH. Add a separate AwaitGlobalnetIP that uses this new method to check for if IP is present or not. That way, you can also test use cases which require a globelIngressIP but allocatedIP to be empty.

Reply:
Can you tell me how I can test a e2e scenario where GlobalIngressIP Object is created but IPAddress is not allocated based on the current implementation?

Reply:
IP allocation failed. You can cause such a failure by creating an EgressIPObject that uses up all the remaining IPs in the pool.

Reply:
As discussed we are validating this in our unit tests and e2e may not be appropriate